packages/ui/src/components/sign-up/sign-up.tsx

```diff
@@ -76,114 +79,89 @@ function SignUpComponentLoaded() {
                         return <Alert>{message}</Alert>;
                       }}
                     </Common.GlobalError>
-                    <Connection.Root>
-                      {enabledConnections.map(c => {
-                        const connection = PROVIDERS.find(provider => provider.id === c.provider);
-                        const iconKey = connection?.icon;
-                        const IconComponent = iconKey ? Icon[iconKey] : null;
-                        return (
-                          <Common.Loading
-                            key={c.provider}
-                            scope={`provider:${c.provider}`}
-                          >
-                            {isConnectionLoading => {
-                              return (
-                                <Common.Connection
-                                  name={c.provider}
-                                  asChild
-                                >
-                                  <Connection.Button
-                                    busy={isConnectionLoading}
-                                    disabled={isGlobalLoading || isConnectionLoading}
-                                    icon={IconComponent ? <IconComponent className='text-base' /> : null}
-                                    textVisuallyHidden={enabledConnections.length > 2}
-                                  >
-                                    {connection?.name || c.provider}
-                                  </Connection.Button>
-                                </Common.Connection>
-                              );
-                            }}
-                          </Common.Loading>
-                        );
-                      })}
-                    </Connection.Root>
 
-                    <Seperator>{t('dividerText')}</Seperator>
+                    <Connections loading={isGlobalLoading} />
```

Comment:
In theory, we could also just move isGlobalLoading into the connections component too if we wanted?